### PR TITLE
Add Inline Blocks Docs

### PR DESCRIPTION
**Pull request by kendallstrautman**
- [x] Add images & gifs
- [x] Proofread / edit
- [x] Add consumer data

Would also still love to do some more testing with external repos and the tinacms.org site. ~~Just today saw some more potential style issues with focus states and such. Might be best to keep these on the backburner.~~ [This PR](https://github.com/tinacms/tinacms/pull/949) should fix that.

### CODE REVIEW

**Comment by kendallstrautman** [comment]
Write a better explanation of how `BlockImage` behaves with/without children

File: content/docs/inline-blocks.md [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+        status === 'active' ? deactivate() : activate()
+      }}
+    >
+      {status === 'active' ? 'Preview' : 'Edit'}
+    </TinaButton>
+  )
+}
+```
+
+**Note:** The above example imports button styles from `@tinacms/styles`, but this component could be a plain `button` element with it's own styles.
```
Comment (DirtyF) [inline content/docs/inline-blocks.md]:
```suggestion
**Note:** The above example imports button styles from `@tinacms/styles`, but this component could be a plain `button` element with its own styles.
```

File: content/docs/inline-blocks/block-image.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  name: string
+  parse(filename: string): string
+  uploadDir(form: Form): string
+  children?: any
+}
+```
+
+## Examples
+
+`BlockImage` will also **pass it's children**. Below is an example of how you could use this with _Gatsby Image_.
```
Comment (DirtyF) [inline content/docs/inline-blocks/block-image.md]:
```suggestion
`BlockImage` will also **pass its children**. Below is an example of how you could use this with _Gatsby Image_.
```

File: content/docs/inline-blocks/block-text.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    <BlocksControls index={props.index}>
+      <h2>
+        <BlockText name="tagline" />
+      </h2>
+    </BlocksControls>
+  )
+}
+```
+
+> **Note**: Block Field [styles can be extended](https://tinacms.org/docs/inline-blocks#extending-block-field-styles) or overridden via _Styled Components_.
```
Comment (DirtyF) [inline content/docs/inline-blocks/block-text.md:36]:
```suggestion
> **Note**: Block Field [styles can be extended](/docs/inline-blocks#extending-block-field-styles) or overridden via _Styled Components_.
```

Comment (DirtyF) [reply content/docs/inline-blocks/block-text.md:36]: Just to make sure the links also works on the preview.

File: content/docs/inline-blocks/block-textarea.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    description: Shows BlockTextarea examples
+  - file: /packages/react-tinacms-inline/src/inline-field-textarea.tsx
+    description: Depends on inline textarea field config
+---
+
+Inline `BlockTextarea` represents a multiline text input. It should be used for content values that are long strings: for example, a page description.
+
+## Definition
+
+Below is an example of how `BlockTextarea` may be used in a [Block Component](https://tinacms.org/docs/inline-blocks#block-component) definition.
```
Comment (DirtyF) [inline content/docs/inline-blocks/block-textarea.md:18]:
```suggestion
Below is an example of how `BlockTextarea` may be used in a [Block Component](/docs/inline-blocks#block-component) definition.
```

File: content/docs/inline-editing.md [hunk 5]
```diff
@@ -95,26 +95,94 @@ export function Page(props) {
...
 > Note that we switched the call to `useLocalForm` with a call to `useForm`. This will prevent the form from showing in the Tina sidebar. If you want the form to also be available in the sidebar, `useForm` can be replaced with `useLocalForm`.
 
-## Activating Edit Mode for Inline Forms
+## Inline Form Controls
+
+There are a few fundamental editing actions needed to handle the state of _Inline Form_: Activating and deactivating 'edit mode', along with saving and discarding changes. Below are some examples of how to manually add these control buttons within an _Inline Form_.
+
+![TinaCMS: Inline Controls](/img/inline-blocks/inline-controls.png)
+
+> This configuration **may differ based on the project**. With the introduction of [Open Authoring](https://tinacms.org/blog/introducing-visual-open-authoring), these editing states are handled through a _Global Toolbar_. The below **implementations may change** as new features are added to the Inline Editing API.
```
Comment (DirtyF) [inline content/docs/inline-editing.md:104]:
```suggestion
> This configuration **may differ based on the project**. With the introduction of [Open Authoring](/blog/introducing-visual-open-authoring), these editing states are handled through a _Global Toolbar_. The below **implementations may change** as new features are added to the Inline Editing API.
```

File: content/docs/inline-blocks.md [hunk 6]
```diff
@@ -0,0 +1,315 @@
...
+    description: Shows Block interface
+  - file: /packages/@tinacms/fields/src/plugins/BlocksFieldPlugin.tsx
+    description: Shows Block Template interface
+  - file: /packages/react-tinacms-inline/src/blocks/inline-field-blocks.tsx
+    description: Shows Inline Blocks interface and gives example
+  - file: /packages/react-modals/src/ModalProvider.tsx
+    description: Uses ModalProvider for Block Settings
+---
+
+Inline Blocks combine the content modelling flexibility of regular [Blocks](https://tinacms.org/blog/what-are-blocks) with the improved editing experience of [Inline Editing](https://tinacms.org/docs/inline-editing).
```
Comment (DirtyF) [inline content/docs/inline-blocks.md:18]:
```suggestion
Inline Blocks combine the content modelling flexibility of regular [Blocks](/blog/what-are-blocks) with the improved editing experience of [Inline Editing](/docs/inline-editing).
```

File: content/docs/inline-blocks.md [hunk 7]
```diff
@@ -0,0 +1,315 @@
...
+
+> **Please note:** This is considered an unstable API. Any breaking changes will be shared in the weekly [Release Notes](https://tinacms.org/blog). That said, we will do our best to keep this document up-to-date.
```
Comment (DirtyF) [inline content/docs/inline-blocks.md:20]:
```suggestion
> **Please note:** This is considered an unstable API. Any breaking changes will be shared in the weekly [Release Notes](/blog). That said, we will do our best to keep this document up-to-date.
```

File: content/docs/inline-blocks.md [hunk 8]
```diff
@@ -0,0 +1,315 @@
...
+
+![TinaCMS: Inline Block Controls](/img/inline-blocks/block-controls.png)
+
+The image above shows the _Textarea Block_ field in use.
+
+**Available Block Fields**:
+
+- [Text Block](https://tinacms.org/docs/inline-blocks/block-text)
+- [Textarea Block](https://tinacms.org/docs/inline-blocks/block-textarea)
+- [Image Block](https://tinacms.org/docs/inline-blocks/block-image)
```
Comment (DirtyF) [inline content/docs/inline-blocks.md:69]:
```suggestion
- [Text Block](/docs/inline-blocks/block-text)
- [Textarea Block](/docs/inline-blocks/block-textarea)
- [Image Block](/docs/inline-blocks/block-image)
```

File: content/docs/inline-blocks.md [hunk 9]
```diff
@@ -0,0 +1,315 @@
...
+  defaultItem: {
+    _template: 'heading',
+    text: 'At vero eos et accusamus',
+  },
+  key: 'heading-block',
+  fields: [],
+}
+```
+
+The _Inline Block Template_ **configures the block** with the CMS. It has a similar shape to a Regular [Block Template](https://tinacms.org/docs/fields/blocks#block-template-options) definition.
```
Comment (DirtyF) [inline content/docs/inline-blocks.md:86]:
```suggestion
The _Inline Block Template_ **configures the block** with the CMS. It has a similar shape to a Regular [Block Template](/docs/fields/blocks#block-template-options) definition.
```

File: content/docs/inline-blocks.md [hunk 10]
```diff
@@ -0,0 +1,315 @@
...
+| `label`        |         `string`          |                                                                              A human readable label. |
+| `defaultItem?` | `object | (() => object)` |                                                              Populates new blocks with default data. |
+| `key`          |         `string`          | A unique value to optimize the [rendering of the list](https://reactjs.org/docs/lists-and-keys.html) |
+| `fields`       |          `Array`          |                                    Populates fields in the [Settings Modal](link to settings modal). |
+
+## Configuring Inline Blocks with Inline Form
+
+The initial steps to configuring _Inline Blocks_ involve setting up an [_Inline Form_](https://tinacms.org/docs/inline-editing#inlineform-and-inlinefield) on the page or component where the blocks should render. Then, you should [add controls](https://tinacms.org/docs/inline-editing#inline-form-controls) to handle editing state. Finally, you can use a component called `InlineBlocks` that **renders blocks in order** based on the source data.
```
Comment (DirtyF) [inline content/docs/inline-blocks.md:98]:
```suggestion
The initial steps to configuring _Inline Blocks_ involve setting up an [_Inline Form_](/docs/inline-editing#inlineform-and-inlinefield) on the page or component where the blocks should render. Then, you should [add controls](/docs/inline-editing#inline-form-controls) to handle editing state. Finally, you can use a component called `InlineBlocks` that **renders blocks in order** based on the source data.
```

File: content/docs/inline-blocks.md [hunk 11]
```diff
@@ -0,0 +1,315 @@
...
+
+### The Steps:
+
+1. Wrap your component with `InlineForm`, pass the `form` object.
+2. Set up [Inline Controls](https://tinacms.org/docs/inline-editing#inline-form-controls).
```
Comment (DirtyF) [inline content/docs/inline-blocks.md:103]:
```suggestion
2. Set up [Inline Controls](/docs/inline-editing#inline-form-controls).
```

File: content/docs/inline-blocks.md [hunk 12]
```diff
@@ -0,0 +1,315 @@
...
+
+## Using the Settings Modal
+
+There may be times when a _Block_ **needs more field inputs** than a single inline field. For example, an image may need a field for the ‘alt’ tag. For this metadata, you can use `ModalProvider` and add additional fields to the _Block Template_.
+
+![TinaCMS: Inline Blocks Settings Modal](/img/inline-blocks/settings-modal.png)
+
+### 1. Define Fields in the Block Template
+
+Field definitions added to the `fields` property will render in the _Settings Modal_. These fields are defined exactly the same as regular [sidebar fields](https://tinacms.org/docs/fields).
```
Comment (DirtyF) [inline content/docs/inline-blocks.md:214]:
```suggestion
Field definitions added to the `fields` property will render in the _Settings Modal_. These fields are defined exactly the same as regular [sidebar fields](/docs/fields).
```

File: content/docs/inline-blocks/block-textarea.md [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+    <BlocksControls index={index}>
+      <h3>
+        <BlockTextarea name="support_copy" />
+      </h3>
+    </BlocksControls>
+  )
+}
+```
+
+> **Note**: Block Field [styles can be extended](https://tinacms.org/docs/inline-blocks#extending-block-field-styles) or overridden via _Styled Components_.
```
Comment (DirtyF) [inline content/docs/inline-blocks/block-textarea.md:36]:
```suggestion
> **Note**: Block Field [styles can be extended](/docs/inline-blocks#extending-block-field-styles) or overridden via _Styled Components_.
```

File: content/docs/inline-blocks/block-image.md [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+```typescript
+export interface InlineImageProps {
+  name: string
+  parse(filename: string): string
+  uploadDir(form: Form): string
+  children?: any
+}
+```
+
+## Examples
```
Comment (DirtyF) [inline content/docs/inline-blocks/block-image.md:62]:
```suggestion
## Example
```

File: content/docs/inline-blocks.md (hunk 6 above)
Comment (kendallstrautman) [inline content/docs/inline-blocks.md:18]:
```suggestion
Inline Blocks combine the content modelling flexibility of regular [Blocks](/blog/what-are-blocks) with the improved editing experience of [Inline Editing](/docs/inline-editing).
```

**Pull request merged**